packages/core/src/modules/dids/methods/peer/peerDidNumAlgo2.ts

```diff
@@ -82,7 +89,7 @@ export function didToNumAlgo2DidDocument(did: string) {
         // FIXME: the peer did uses key identifiers without the multi base prefix
         // However method 0 (and thus did:key) do use the multi base prefix in the
         // key identifier. Fixing it like this for now, before making something more complex
-        verificationMethod.id = verificationMethod.id.replace('#z', '#')
+        // verificationMethod.id = verificationMethod.id.replace('#z', '#')
```

Comment:
It looks like there is a contradiction between the two specifications:
- In [Peer DID spec](https://identity.foundation/peer-did-method-spec/#multi-key-creation) DIDs doesn't have z letter in the did-url (did fragment after #).
- In [DIDComm spec](https://identity.foundation/didcomm-messaging/spec/#:~:text=%22id%22%3A%20%22did%3Aexample%3A123%23zC9ByQ8aJs8vrNXyDhPHHNNMSHPcaSgNpjjsBYpMMjsTdS%22%2C) DIDs have z letter in the did-url

In our understanding, `z` prefix should always be used for multi-base key representation which is used.
Sicpa `didcomm-rust` requestes Peer DIDs with the additional letter z at the start of the did-url when Peer DID itself doesn't have it.
So, now I commented the code to get compatibility with the didcomm-rust library. It you consider this wrong behaviour we will need to change `didcomm-rust`

Reply:
The DIDComm spec uses did:example and is just to indicate a reference in the did document. It's not based on a specification. The Peer DID specification however defines exactly how you need to generate key ids for the did:peer method. 

As we're creating did:peer dids here the Peer DID Spec is leading, and we should either follow the specification, or update the specification if it is indeed incorrect. AFJ just follows what is described in the specification.

> In our understanding, z prefix should always be used for multi-base key representation which is used.

I don't think there's such a rule on key identifiers  for generic dids.

> So, now I commented the code to get compatibility with the didcomm-rust library. It you consider this wrong behaviour we will need to change didcomm-rust

I find it odd we don't prepend the z prefix, but It's what is described by the specification and thus also what others implement (e.g. the did:peer implementation by TBD: https://github.com/TBD54566975/ssi-sdk/blob/main/did/peer_test.go#L226). So, yes, I think the didcomm-rust library needs to be changed.

Reply:
We found an issue inside our cages with caused us to do this change. So, now we reverted the above change.

Reply:
Got rid of the above replacement `#z` with `#` and implemented separate resolution logics for key DIDs and for peer DIDs (method 0 and method 2) in scope of a separate PR: https://github.com/hyperledger/aries-framework-javascript/pull/1148.

Please take a note that it will actually change the resolution logic for peer DIDs **method 0**. Previously the resulting key fragment IDs for `did:peer:0` included the multibase prefix (the same as for `did:key`) but now it will omit it (the same as for `did:peer:2`).